contributingGuides/TS_STYLE.md

```diff
@@ -0,0 +1,515 @@
+
+  ```ts
+  // Most simple form of union type.
+  type Color = "red" | "green" | "blue";
+  function printColors(color: Color) {
+    console.log(color);
+  }
+
+  // When the values need to be iterated upon.
+  import { TupleToUnion } from "type-fest";
+
+  const COLORS = ["red", "green", "blue"] as const;
+  type Color = TupleToUnion<typeof COLORS>; // type: 'red' | 'green' | 'blue'
+
+  for (const colors of color) {
+    printColor(color);
+  }
+
+  // When the values should be accessed through object keys. (i.e. `COLORS.Red` vs. `"red"`)
+  import { ValueOf } from "type-fest";
+
+  const COLORS = {
+    Red: "red",
+    Green: "green",
+    Blue: "blue",
+  } as const;
+  type Color = ValueOf<typeof COLORS>; // type: 'red' | 'green' | 'blue'
+
+  printColor(COLORS.Red);
+  ```
+
+<a name="unknown-vs-any"></a><a name="1.5"></a>
+
+- [1.5](#unknown-vs-any) **`unknown` vs. `any`**: Don't use `any`. Use `unknown` if type is not known beforehand. eslint: [`@typescript-eslint/no-explicit-any`](https://typescript-eslint.io/rules/no-explicit-any/)
+
+  > Why? `any` type bypasses type checking. `unknown` is type safe as `unknown` type needs to be type narrowed before being used.
+
+  ```ts
+  const value: unknown = JSON.parse(someJson);
+  if (typeof value === 'string') {...}
+  else if (isPerson(value)) {...}
+  ...
+  ```
+
+<a name="array"></a><a name="1.6"></a>
+
+- [1.6](#array) **`T[]` vs. `Array<T>`**: Use `T[]` or `readonly T[]` for simple types (i.e. types which are just primitive names or type references). Use `Array<T>` or `ReadonlyArray<T>` for all other types (union types, intersection types, object types, function types, etc). eslint: [`@typescript-eslint/array-type`](https://typescript-eslint.io/rules/array-type/)
+
+  ```ts
+  // Array<T>
+  const a: Array<string | number> = ["a", "b"];
+  const b: Array<{ prop: string }> = [{ prop: "a" }];
+  const c: Array<() => void> = [() => {}];
+
+  // T[]
+  const d: MyType[] = ["a", "b"];
+  const e: string[] = ["a", "b"];
+  const f: readonly string[] = ["a", "b"];
+  ```
+
+<a name="ts-ignore"></a><a name="1.7"></a>
+
+- [1.7](#ts-ignore) **@ts-ignore**: Do not use `@ts-ignore` or its variant `@ts-nocheck` to suppress warnings and errors.
+
+  > Use `@ts-expect-error` during the migration for type errors that should be handled later. Refer to the [Migration Guidelines](#migration-guidelines) for specific instructions on how to deal with type errors during the migration. eslint: [`@typescript-eslint/ban-ts-comment`](https://typescript-eslint.io/rules/ban-ts-comment/)
+
+<a name="ts-nullish-coalescing"></a><a name="1.8"></a>
+
+- [1.8](#ts-nullish-coalescing) **Optional chaining and nullish coalescing**: Use optional chaining and nullish coalescing instead of the `get` lodash function. eslint: [`no-restricted-imports`](https://eslint.org/docs/latest/rules/no-restricted-imports)
```

Comment:
```suggestion
- [1.8](#ts-nullish-coalescing) **Optional chaining and nullish coalescing**: Use optional chaining and nullish coalescing instead of the `get` lodash function. ESLint rule: [`no-restricted-imports`](https://eslint.org/docs/latest/rules/no-restricted-imports)
```